Terminal-iOS/MyStudy/StudyDetail/ViewComponent/AlertView/TerminalAlertUIView.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  TerminalAlertUIView.swift
+//  Terminal-iOS
+//
+//  Created by 정재인 on 2021/01/05.
+//  Copyright © 2021 정재인. All rights reserved.
+//
+
+import UIKit
+
+class TerminalAlertUIView: UIView {
+    var redButton = UIButton()
+    var yellowButton = UIButton()
+    var greenButton = UIButton()
+    var topBar = UIView()
+    var bottomBar = UIView()
+    var guideLabel = UILabel()
+    var dismissButton = UIButton()
+    var completeButton = UIButton()
+    
+    init() {
+        super.init(frame: CGRect.zero)
+        layout()
+        attribute()
+    }
+    
+    func attribute() {
+        self.do {
+            $0.backgroundColor = UIColor.appColor(.alertBackgroundColor)
+            $0.layer.cornerRadius = 5
+            $0.layer.masksToBounds = true
+        }
+        topBar.do {
+            $0.backgroundColor = UIColor.appColor(.alertTopBarColor)
+            $0.layer.cornerRadius = 5
+            $0.layer.masksToBounds = true
+        }
+        bottomBar.do {
+            $0.backgroundColor = UIColor.appColor(.alertBackgroundColor)
+        }
+        redButton.do {
+            $0.layer.masksToBounds = true
+            $0.layer.cornerRadius = 5
+            $0.backgroundColor = UIColor.appColor(.redButtonColor)
+            $0.setTitle("", for: .normal)
+        }
+        yellowButton.do {
+            $0.layer.masksToBounds = true
+            $0.layer.cornerRadius = 5
+            $0.backgroundColor = UIColor.appColor(.yellowButtonColor)
+            $0.setTitle("", for: .normal)
+        }
+        greenButton.do {
+            $0.layer.masksToBounds = true
+            $0.layer.cornerRadius = 5
+            $0.backgroundColor = UIColor.appColor(.greenButtonColor)
+            $0.setTitle("", for: .normal)
+        }
```

Comment:
기존 기획과는 별개로 컨벤션에 대한 이야기인데 

클래스 내부에서는 명시적으로 self를 사용해주는 방향에 대해서 어떻게 생각하나요??

예를 들면 이런식으로 말이죠

```suggestion
        self.greenButton.do {
            $0.layer.masksToBounds = true
            $0.layer.cornerRadius = 5
            $0.backgroundColor = UIColor.appColor(.greenButtonColor)
            $0.setTitle("", for: .normal)
        }
```

Reply:
음 더 생각해봐야할 것 같은데 
https://velog.io/@co-in/%EA%B3%B5%EC%8B%9D-%EB%AC%B8%EC%84%9C%EB%A1%9C-%EA%B3%B5%EB%B6%80%ED%95%98%EB%8A%94-Swift-10-%EB%A9%94%EC%86%8C%EB%93%9C
공식 문서 해석본에도 자주사용할 필요가 없다, 쓰지 않는다면 해당 인스턴스의 프로퍼티나 메소드로 간주한다 라고 되어있어서
서 저희끼리 필요한 이유가 있다면 이유를 정확히 하고 쓰면 좋을 것 같아요